pkgs/build-support/rust/hooks/cargo-check-hook.sh

```diff
@@ -28,14 +28,18 @@ cargoCheckHook() {
         cargoCheckFeaturesFlag="--features=${cargoCheckFeatures// /,}"
     fi
 
-    argstr="${cargoCheckProfileFlag} ${cargoCheckNoDefaultFeaturesFlag} ${cargoCheckFeaturesFlag}
-        --target @rustHostPlatformSpec@ --offline ${cargoTestFlags}"
-
     (
         set -x
         cargo test \
               -j $NIX_BUILD_CORES \
-              ${argstr} -- \
+              ${cargoCheckProfileFlag} \
+              ${cargoCheckNoDefaultFeaturesFlag} \
+              ${cargoCheckFeaturesFlag} \
+              --target @rustHostPlatformSpec@ \
+              --offline \
+              ${cargoTestFlags} \
+              ${cargoTestFlagsArray+"${cargoTestFlagsArray[@]}"} \
```

Comment:
why not `"${cargoTestFlagsArray[@]}"`?

Reply:
Just copied that pattern from `checkFlagsArray` further down. My understanding is that they are functionally identical though so I could probably simplify this a bit.

Reply:
it's fine then